jenkins/deploy-clio-server.sh

```diff
@@ -0,0 +1,206 @@
+#             the docker image
+
+# verify required vars are set to actual values
+if [ -z "$ENV" ]
+then
+   echo "Error: ENV not set!!"
+   exit 2
+fi
+
+if [ -z "$CLIO_HOST" ]
+then
+   echo "Error: CLIO_HOST not set!!"
+   exit 2
+fi
+
+CONFIG_DIR=clio-server/config
+
+if [ ! -d "$CONFIG_DIR" ]
+then
+    echo "Missing configuration directory ($CONFIG_DIR) - Exitting "
+    exit 1
+fi
+
+# set VAULT token location
+if [ -f /etc/vault-token-dsde ]
+then
+   TOKEN_FILE="/etc/vault-token-dsde"
+elif [ -f ${HOME}/.vault-token ]
+then
+   TOKEN_FILE="${HOME}/.vault-token"
+else
+   echo "Missing VAULT TOKEN file - Exitting"
+   exit 1
+fi
+   
+# initialize vars
+# location on host where configs should get copied
+APP_DIR=${APP_DIR:-"/app"}
+# mappings from files on the host VM to files in the docker
+HOST_LOG_DIR=/local/clio_logs
+CONTAINER_LOG_DIR=/logs
+HOST_CLIO_CONF=${APP_DIR}/clio.conf
+CONTAINER_CLIO_CONF=/etc/clio.conf
+HOST_CLIO_LOGBACK=${APP_DIR}/clio-logback.xml
+CONTAINER_CLIO_LOGBACK=/etc/clio-logback.xml
+HOST_CLIO_PORT=80
+CONTAINER_CLIO_PORT=8080
+
+# SSH options
+#SSH_USER=jenkins
+SSH_USER=jenkins
+SSHOPTS="-o UserKnownHostsFile=/dev/null -o CheckHostIP=no -o StrictHostKeyChecking=no"
+SSHCMD="ssh $SSHOPTS"
+SCPCMD="scp $SSHOPTS"
+
+# sbt command to disable tests in the docker build
+if [[ "$RUN_TESTS" = "false" ]]; then
+    SET_TESTS="set test in assembly in LocalProject(\"clio-server\") := {}"
+fi
+# sbt command to override the default docker tag
+if [[ ! -z "$DOCKER_TAG" ]]; then
+    SET_VERSION="set version := \"$DOCKER_TAG\""
+fi
+
+# build the clio-server docker image
+docker run --rm \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+    -v ${CLIO_DIR}:/clio \
+    -w="/clio" \
+    broadinstitute/scala:scala-2.11.8 sbt "$SET_TESTS" "$SET_VERSION" "clio-server/docker"
+
+# if no tag was given, read the default tag from the generated resource file
+if [[ -z "$DOCKER_TAG" ]]; then
+	DOCKER_TAG=$(cat ${CLIO_DIR}/clio-server/target/scala-2.12/resource_managed/main/clio-server-version.conf | perl -pe 's/clio\.server\.version:\s+//g')
+fi
+
+# push the docker image
+docker push broadinstitute/clio-server:${DOCKER_TAG}
+
+# TMPDIR
+TMPDIR=$(mktemp -d ${progname}-XXXXXX)
+# temporary env file for rendering ctmpls
+ENV_FILE="${TMPDIR}/env-vars.txt"
+
+# populate temp env.txt file with necessary environment
+echo "ENVIRONMENT=${ENV}" >> ${ENV_FILE}
+echo "ENV=${ENV}" >> ${ENV_FILE}
+echo "DOCKER_TAG=${DOCKER_TAG}" >> ${ENV_FILE}
+echo "HOST_LOG_DIR=${HOST_LOG_DIR}" >> ${ENV_FILE}
+echo "CONTAINER_LOG_DIR=${CONTAINER_LOG_DIR}" >> ${ENV_FILE}
+echo "HOST_CLIO_CONF=${HOST_CLIO_CONF}" >> ${ENV_FILE}
+echo "CONTAINER_CLIO_CONF=${CONTAINER_CLIO_CONF}" >> ${ENV_FILE}
+echo "HOST_CLIO_LOGBACK=${HOST_CLIO_LOGBACK}" >> ${ENV_FILE}
+echo "CONTAINER_CLIO_LOGBACK=${CONTAINER_CLIO_LOGBACK}" >> ${ENV_FILE}
+echo "HOST_CLIO_PORT=${HOST_CLIO_PORT}" >> ${ENV_FILE}
+echo "CONTAINER_CLIO_PORT=${CONTAINER_CLIO_PORT}" >> ${ENV_FILE}
```

Comment:
Having all these variables might be overboard, especially for the ones that don't need to be used in multiple templates.